### ISSUE TITLE
Adding multiple OpenShift clusters

### ISSUE THREAD

**Issue by cooktheryan** [argoproj/argo-cd#1761]
**Describe the bug**
When attempting to add a second OpenShift cluster a secret must be deleted.

**To Reproduce**
If we cannot reproduce, we cannot fix! Steps to reproduce the behavior:
```
[rcook@goku git-fork]$ argocd cluster add west2 --kubeconfig=/home/rcook/git/pit-hybrid/kube-config/aws-east1-east2-west2
INFO[0000] ServiceAccount "argocd-manager" created      
INFO[0000] ClusterRole "argocd-manager-role" created    
INFO[0000] ClusterRoleBinding "argocd-manager-role-binding" created, bound "argocd-manager" to "argocd-manager-role" 
FATA[0001] Secret "argocd-manager-dockercfg-z8l9j" for service account "&ServiceAccount{ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{Name:argocd-manager,GenerateName:,Namespace:kube-system,SelfLink:/api/v1/namespaces/kube-system/serviceaccounts/argocd-manager,UID:57d65209-9138-11e9-84d2-0a5b89ddb17e,ResourceVersion:4817202,Generation:0,CreationTimestamp:2019-06-17 15:44:38 -0400 EDT,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,},Secrets:[{  argocd-manager-dockercfg-z8l9j    } {  argocd-manager-token-8scvq    }],ImagePullSecrets:[{argocd-manager-dockercfg-z8l9j}],AutomountServiceAccountToken:nil,}" did not have a token 
[rcook@goku git-fork]$ oc delete secret argocd-manager-dockercfg-z8l9j --context west2 -n kube-system
secret "argocd-manager-dockercfg-z8l9j" deleted
[rcook@goku git-fork]$ argocd cluster add west2 --kubeconfig=/home/rcook/git/pit-hybrid/kube-config/aws-east1-east2-west2
INFO[0000] ServiceAccount "argocd-manager" already exists 
INFO[0000] ClusterRole "argocd-manager-role" updated    
INFO[0000] ClusterRoleBinding "argocd-manager-role-binding" already exists 
```

**Expected behavior**
We shouldn't need to delete secret but there is a nice workaround.

**Screenshots**

**Version**

```shell
argocd: v1.0.1+5fe1447.dirty
  BuildDate: 2019-05-28T17:26:35Z
  GitCommit: 5fe1447b722716649143c63f9fc054886d5b111c
  GitTreeState: dirty
  GoVersion: go1.11.4
  Compiler: gc
  Platform: linux/amd64
argocd-server: v1.1.0-rc1+46550e0
  BuildDate: 2019-06-14T18:27:01Z
  GitCommit: 46550e009bfe4525cd6399a631ded8c9eb34f02e
  GitTreeState: clean
  GoVersion: go1.11.4
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: 0.13.1

```

**Logs**

```
Paste any relevant application logs here.
```

**Have you thought about contributing a fix yourself?**

Open Source software thrives with your contribution. It not only gives skills you might not be able to get in your day job, it also looks amazing on your resume. 

If you want to get involved, check out the 
[contributing guide](https://github.com/argoproj/argo-cd/blob/master/docs/CONTRIBUTING.md), then reach out to us on [Slack](https://argoproj.github.io/community/join-slack) so we can see how to get you started.

**Comment by alexec** [comment]
Thank you. Can I ask what the work-around is please?

**Comment by cooktheryan** [comment]
sorry it was in the paste. So the only thing required it seems to delete the secret and then run the command again

```
argocd cluster add west2 --kubeconfig=/home/rcook/git/pit-hybrid/kube-config/aws-east1-east2-west2
oc delete secret argocd-manager-dockercfg-z8l9j --contex west2 -n kube-system
argocd cluster add west2 --kubeconfig=/home/rcook/git/pit-hybrid/kube-config/aws-east1-east2-west2
```

**Comment by alexec** [comment]
thank you

**Comment by christianh814** [comment]
Ran into this today too. I wonder what the issue is? Workaround is fine, but why does it fail in the first place?

**Comment by christianh814** [comment]
Just and update ....this still happens from time to time and I'm not sure why.

**Comment by dgoodwin** [comment]
Based on: "FATA[0001] Secret "argocd-manager-dockercfg-z8l9j" for service account" I suspect this was fixed with: https://github.com/argoproj/argo-cd/pull/2744

Bug was Argo assuming the first service account token secret it found was the right one. This is often not true in openshift where there are other service account secrets.

This was fixed in 1.4.x I believe, but *not* 1.3.x. I would suggest closing this issue unless someone can reproduce on 1.4.x.

**Comment by alexmt** [comment]
Thank you @dgoodwin , closing the issue.

 @christianh814, @cooktheryan  please let us know if you still can reproduce the problem.
